public/app/plugins/panel/flamegraph/components/FlameGraph/FlameGraph.tsx

```diff
@@ -70,138 +86,88 @@ const FlameGraph = ({
 
   const [clickedItemData, setClickedItemData] = useState<ClickedItemData>();
 
-  const [ufuzzy] = useState(() => {
-    return new uFuzzy();
-  });
-
-  const foundLabels = useMemo(() => {
-    const foundLabels = new Set<string>();
-
-    if (search) {
-      let idxs = ufuzzy.filter(data.getUniqueLabels(), search);
-
-      if (idxs) {
-        for (let idx of idxs) {
-          foundLabels.add(data.getUniqueLabels()[idx]);
-        }
-      }
-    }
-
-    return foundLabels;
-  }, [ufuzzy, search, data]);
-
-  useEffect(() => {
-    if (!levels.length) {
-      return;
-    }
-    const pixelsPerTick = (wrapperWidth * window.devicePixelRatio) / totalTicks / (rangeMax - rangeMin);
-    const ctx = graphRef.current?.getContext('2d')!;
-    const graph = graphRef.current!;
-
-    const height = PIXELS_PER_LEVEL * levels.length;
-    graph.width = Math.round(wrapperWidth * window.devicePixelRatio);
-    graph.height = Math.round(height * window.devicePixelRatio);
-    graph.style.width = `${wrapperWidth}px`;
-    graph.style.height = `${height}px`;
+  useFlameRender(
+    graphRef,
+    data,
+    levels,
+    wrapperWidth,
+    rangeMin,
+    rangeMax,
+    search,
+    textAlign,
+    totalTicks,
+    focusedItemData
+  );
 
-    ctx.textBaseline = 'middle';
-    ctx.font = 12 * window.devicePixelRatio + 'px monospace';
-    ctx.strokeStyle = 'white';
+  const onGraphClick = useCallback(
```

Comment:
Some small changes but mainly I added the handlers as a react prop instead of manually adding the handlers in the effect.